monty-31/src/aarch64_neon/packing.rs

```diff
@@ -766,3 +766,24 @@ pub(crate) fn base_mul_packed<FP, const WIDTH: usize>(
         _ => panic!("Unsupported binomial extension degree: {}", WIDTH),
     }
 }
+
+/// Raise MontyField31 field elements to a small constant power `D`.
+///
+/// `D` must be one of 3, 5, or 7, otherwise this function will panic at compile time.
+///
+/// # Safety
+/// Inputs must be signed 32-bit integers in the range `[-P, P)`.
```

Comment:
We can work with a slightly more general range:
```suggestion
/// Inputs must be signed 32-bit integers in the range `[-P, P]`.
```

Reply:
Can you also go through and update the ranges for the inputs for `mul, mul_with_precomp, cube, exp_5, exp_7` from `(-P, P)` to `[-P, P]` so everything matches.